### PR TITLE
[MM-55595] Use annotated logger in search layer

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
To debug issues with the search layer, i.e. Bleve and Elasticsearch, more easily, the warning and error logged at that level should contain the full context of their origination, i.e., use the logger from `request.CTX`.

That required a lot of changes as the calls are deep down in the stack. I tried to keep the changes reasonable, but it's still a lot. 

Several methods like `UserStore.Search` are called in so many places that it's unreasonable to add another argument to their signature as part of this PR. I will file follow-up tickets to address that later.

One notable change is that `sqlstore.Store` now has its own `Logger`. `CreateConsoleLogger()` was added to `mlog` to support a logger creation place where no configuration and no `testing.T` is accessible. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-55595

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/store/sqlstore/adapters.go [hunk 1]
```diff
@@ -50,16 +46,6 @@ func (str jsonKeyPath) Value() (driver.Value, error) {
...
 
-type TraceOnAdapter struct{}
-
-func (t *TraceOnAdapter) Printf(format string, v ...any) {
-	originalString := fmt.Sprintf(format, v...)
-	newString := strings.ReplaceAll(originalString, "\n", " ")
-	newString = strings.ReplaceAll(newString, "\t", " ")
-	newString = strings.ReplaceAll(newString, "\"", "")
-	mlog.Debug(newString)
-}
```
Comment (hanzei) [inline server/channels/store/sqlstore/adapters.go:61]: Unused code

File: server/channels/store/sqlstore/store.go [hunk 2]
```diff
@@ -239,24 +241,30 @@ func New(settings model.SqlSettings, metrics einterfaces.MetricsInterface) (*Sql
...
 		defer cancel()
 		err = db.PingContext(ctx)
 		if err != nil {
 			if i == attempts-1 {
 				return nil, err
 			}
-			mlog.Error("Failed to ping DB", mlog.Err(err), mlog.Int("retrying in seconds", DBPingTimeoutSecs))
+			logger.Error("Failed to ping DB", mlog.Int("retrying in seconds", DBPingTimeoutSecs), mlog.Err(err))
 			time.Sleep(DBPingTimeoutSecs * time.Second)
 			continue
```
Comment (hanzei) [inline server/channels/store/sqlstore/store.go:269]: The changes here should be double-checked, but I think they are fine.

File: server/platform/services/searchengine/interface.go [hunk 3]
```diff
@@ -7,6 +7,7 @@ import (
 	"time"
 
 	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/request"
 )
 
 type SearchEngineInterface interface {
```
Comment (hanzei) [inline server/platform/services/searchengine/interface.go:13]: At some point, I would like to come back to the interface and migrate the rest of the methods to use a `request.CTX`. But that can happen later.

File: server/channels/store/sqlstore/store.go (hunk 2 above)
Comment (isacikgoz) [reply server/channels/store/sqlstore/store.go:269]: Good catch 👍

File: server/channels/store/searchlayer/user_layer.go [hunk 4]
```diff
@@ -51,52 +52,56 @@ func (s *SearchUserStore) Search(teamId, term string, options *model.UserSearchO
...
 
 	if err == nil {
-		s.rootStore.indexUser(userUpdate.New)
+		s.rootStore.indexUser(rctx, userUpdate.New)
 	}
 	return userUpdate, err
 }
 
 func (s *SearchUserStore) Save(user *model.User) (*model.User, error) {
+	//TODO: Use the actuall request context from the App layer
```
Comment (isacikgoz) [inline server/channels/store/searchlayer/user_layer.go:85]: Maybe we can add the jira ticket number to these

File: server/channels/store/sqlstore/preference_store.go [hunk 5]
```diff
@@ -22,7 +22,6 @@ func newSqlPreferenceStore(sqlStore *SqlStore) store.PreferenceStore {
 }
 
 func (s SqlPreferenceStore) deleteUnusedFeatures() {
-	mlog.Debug("Deleting any unused pre-release features")
```
Comment (isacikgoz) [inline server/channels/store/sqlstore/preference_store.go:25]: Why this was removed?

**Review by isacikgoz — APPROVED** [review]
Great work, a couple of non-blocking comments.

File: server/channels/store/searchlayer/user_layer.go (hunk 4 above)
Comment (hanzei) [reply server/channels/store/searchlayer/user_layer.go]: Good idea, linked in 9361a87fb6

**Pull request merged**